contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,204 @@
+
+```
+$("#textbox").numericTextBox({
+    value: 10,
+    min: -10,
+    max: 100,
+    step: 0.75,
+    format: "n",
+    decimals: 3
+});
+```
+
+<box type="info">
+If you are not familiar with jQuery, you may want to get a brief understanding from the official <a href="https://jquery.com/">website</a>.
+</box>
+
+On the other hand, using versions Angular 2 and above, your code would look something like this:
+
+```
+<input numerictextbox k-min="-10" k-max="100" k-step="0.75"
+  k-format="n" k-decimals="3"/>
+```
+
+Rather than spending time thinking on how the program should flow, you define what you want the user to see and Angular will take care of the dependencies.
+
+The above example shows how Angular makes it easier for developers to design the UI elements.
+
+[source](https://www.telerik.com/blogs/three-ds-of-web-development-1-declarative-vs-imperative)
+
+### Benefit 2: Simplified MVC pattern and reduced coding.
+
+Angular framework uses the [MVC](https://blog.angular-university.io/why-angular-angular-vs-jquery-a-beginner-friendly-explanation-on-the-advantages-of-angular-and-mvc/) (Model-View-Controller) structure. The HTML represents the View component. Next to it, we have the `component.ts` files. This is the controller. Essentially, it can choose what data to push to our view (`.html`). 
+Lastly, we have the model. In Angular, the model will mostly be our services, which we can access through our controller. Services are a way to retrieve, update and process data to and from the backend.
+
+![File Structure](file.png)
+
+Most frameworks require developers to split the application into multiple MVC components. After that, the developer has to write code to put them together. However, Angular does not require you to connect these components together. As explained in this [article](https://blog.angular-university.io/why-angular-angular-vs-jquery-a-beginner-friendly-explanation-on-the-advantages-of-angular-and-mvc/) on the official website, everything happens under the hood. Angular ensures automatic synchronization between the components. That saves developers time.
+
+### Benefit 3: Angular CLI
+
+In 2016 the Angular team announced the Angular CLI, a command-line interface (CLI) to automate your development workflow in applications using versions Angular 2 and above.
+It is recommended to use Angular CLI for creating Angular apps as you don't need to spend time installing and configuring all the required dependencies.
+
+> npm install -g @angular/cli
+
+Running the above command installs the Angular CLI and gives you more control over your project.
+With Angular CLI developers can generate Angular files, execute applications and even run end to end tests.
+
+[source](https://www.amadousall.com/why-you-should-use-angular-cli/)
+
+### What does an Angular app look like?
```

Comment:
I think this section should come very early in the article. Otherwise it is like describing the features of an elephant, explaining what it can do, and then showing a photo of the elephant. The photo should come first.